### ISSUE TITLE
What is best Android TV box to use with Moonlight?

### ISSUE THREAD

**Issue by ronferri** [moonlight-stream/moonlight-android#381]
Which Android TV box offers lowest latency and fastest hardware decoding for current version of Moonlight App?

Nvidia Shield TV?

Himedia Q5 Pro?

Xiaomi Mi Box?

Others?

Thank you!

**Comment by thw0rted** [comment]
Did you need to do something special to pair your DS4, @thealex626 ?  I can't get mine to pair with my Fire Stick, and from reading it sounds like I will have to use a utility on my PC to set the paired MAC address on the controller, or something like that.

**Comment by thealex626** [comment]
I didn't test it on earlier versions, but with the 5.2.4.0 firmware on my box it pairs well. I just go to Bluetooth->game controllers, then I hold PS+Share buttons on my DualShock 4 until the light starts pulsing, and when it becomes visible in devices list, I click Select to pair it. Sometimes the bluetooth device won't appear on the list, but it works as intended if you restart your Fire TV.

After the controller had been paired, it works as a usual remote. However the Cross button serves as Back button, and the Square acts as Select button. This is a bit strange and you'll have to get used to it once you get it working. Normally I'd prefer Cross to act as Select, and Circle to act as Back (Just like A/B buttons on Xbox controller). But once you launch Moonlight and open Steam Big Picture, it will work as intended and the normal Xbox bindings will be assigned to your DS4 layout. Unfortunately tochpad will never work, and I already asked about that: https://github.com/moonlight-stream/moonlight-android/issues/371, your DS4 controller will serve as a normal Xbox One controller.

When not used for a long time, controller switches off. To switch it on simply press the PS button - and it will pair with Fire TV almost instantly. 

I use the 2nd generation DS4 controller, but maybe the 1st gen is more difficult to pair, I don't know. Also I never tried DS4 with Fire TV Stick, I only have the Box version. In general the DS4 is in my opinion the best controller for Fire TV/Moonlight.

**Comment by thw0rted** [comment]
This weekend, I checked and my firmware showed as up to date (though I didn't think to make note of the version).  I tried pairing just as you describe and got a problem I've seen described elsewhere online, it just says it's pairing then after a minute or so it times out and says there was an error.  I do have a 2nd gen controller, so I'm not sure if I'm still doing something wrong or if there's actually some kind of hardware problem.  I'll probably put this project on the back burner for a little while...
